orangecontrib/geo/widgets/owchoropleth.py

```diff
@@ -710,6 +711,18 @@ def set_data(self, data):
         self.apply_selection()
         self.unconditional_commit()
 
+    def check_data(self):
+        self.Warning.no_region.clear()
+        self.Warning.no_lat_lon_vars.clear()
+        self._attr_lat, self._attr_lon = None, None
+        if self.data is not None:
+            self._attr_lat, self._attr_lon = \
+                find_lat_lon(self.data, filter_hidden=True)
+            if self._attr_lat is None or self._attr_lon is None:
+                # we either find both or none
+                self.Warning.no_lat_lon_vars()
+                self.data = None
```

Comment:
Maybe also set both to `None` here, so that the widget doesn't keep dangling variables when `data` is `None`?